sdk/storage/azure-storage-blob/azure/storage/blob/_shared/authentication.py

```diff
@@ -0,0 +1,152 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import base64
+import hashlib
+import hmac
+import logging
+import sys
+try:
+    from urllib.parse import urlparse, unquote
+except ImportError:
+    from urlparse import urlparse
+    from urllib2 import unquote
+
+from azure.core.exceptions import ClientAuthenticationError
+from azure.core.pipeline.policies import SansIOHTTPPolicy
+
+if sys.version_info < (3,):
+    _unicode_type = unicode  # pylint: disable=undefined-variable
+else:
+    _unicode_type = str
+logger = logging.getLogger(__name__)
+
+
+def _encode_base64(data):
```

Comment:
Should these methods in authentication.py? In _shared/encryption.py there is a line `from .authentication import _encode_base64, _decode_base64_to_bytes`

Reply:
Yeah - good call - there's a fair amount of duplicate code in the shared module.
I will undertake a proper refactoring of this module for the next preview.
I've opened an issue here:
https://github.com/Azure/azure-sdk-for-python/issues/6030